pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClient.java

```diff
@@ -117,14 +137,17 @@ void checkForIrrecoverableError(Record<?> record, BulkProcessor.BulkOperationRes
                 isMalformed = true;
                 switch (config.getMalformedDocAction()) {
                     case IGNORE:
+                        incrementCounter(METRICS_TOTAL_MALFORMED_IGNORE, 1);
                         break;
                     case WARN:
+                        incrementCounter(METRICS_TOTAL_SKIP, 1);
                         log.warn("Ignoring malformed document index={} id={}",
                                 result.getIndex(),
                                 result.getDocumentId(),
                                 error);
                         break;
                     case FAIL:
+                        incrementCounter(METRICS_TOTAL_FAILURE, 1);
```

Comment:
For me failed == have not been indexed, whether the reason was malformed or not. So we need to decide how to differentiate.
Maybe: es_total_failures, es_skipped_failures

Also, I wouldn't differentiate between IGNORE and WARN. Don't see a good reason for it.

Reply:
In our case, we need to track different ignored cases and skipped. I'm going to make a change to count them as all failures but peg ignores and skip too.